openfe/tests/setup/test_network.py

```diff
@@ -0,0 +1,239 @@
+from dataclasses import dataclass
+from typing import Iterable
+import pytest
+
+from openfe.setup import Molecule, AtomMapping, Network
+from rdkit import Chem
+
+from networkx import NetworkXError
+
+
+@dataclass
+class _NetworkTestContainer:
+    """Container to facilitate network testing"""
+    network: Network
+    nodes: Iterable[Molecule]
+    edges: Iterable[AtomMapping]
+
+    @property
+    def n_nodes(self) -> int:
+        return len(self.nodes)
+
+    @property
+    def n_edges(self) -> int:
+        return len(self.edges)
+
+
+@pytest.fixture
+def mols():
+    mol1 = Molecule(Chem.MolFromSmiles("CCO"))
+    mol2 = Molecule(Chem.MolFromSmiles("CC"))
+    mol3 = Molecule(Chem.MolFromSmiles("CO"))
+    return mol1, mol2, mol3
+
+
+@pytest.fixture
+def std_edges(mols):
+    mol1, mol2, mol3 = mols
+    edge12 = AtomMapping(mol1, mol2, {0: 0, 1: 1})
+    edge23 = AtomMapping(mol2, mol3, {0: 0})
+    edge13 = AtomMapping(mol1, mol3, {0: 0, 2: 1})
+    return edge12, edge23, edge13
+
+
+@pytest.fixture
+def simple_network(mols, std_edges):
+    """Network with no edges duplicated and all nodes in edges"""
+    network = Network(std_edges)
+    return _NetworkTestContainer(
+        network=network,
+        nodes=mols,
+        edges=std_edges,
+    )
+
+
+@pytest.fixture
+def doubled_edge_network(mols, std_edges):
+    """Network with more than one edge associated with a node pair"""
+    mol1, mol2, mol3 = mols
+    extra_edge = AtomMapping(mol1, mol3, {0: 0, 1: 1})
+    edges = list(std_edges) + [extra_edge]
+    return _NetworkTestContainer(
+        network=Network(edges),
+        nodes=mols,
+        edges=edges,
+    )
+
+
+@pytest.fixture
+def singleton_node_network(mols, std_edges):
+    """Network with nodes not included in any edges"""
+    extra_mol = Molecule(Chem.MolFromSmiles("CCC"))
+    all_mols = list(mols) + [extra_mol]
+    return _NetworkTestContainer(
+        network=Network(edges=std_edges, nodes=all_mols),
+        nodes=all_mols,
+        edges=std_edges,
+    )
+
+
+_NETWORK_NAMES = ['simple', 'doubled_edge', 'singleton_node']
+
+
+@pytest.fixture
+def all_networks(
+    simple_network,
+    doubled_edge_network,
+    singleton_node_network,
+):
+    """Fixture to allow parameterization of the network test"""
+    network_dct = {
+        'simple': simple_network,
+        'doubled_edge': doubled_edge_network,
+        'singleton_node': singleton_node_network,
+    }
```

Comment:
This looks like the problem of trying to have a fixture of fixtures, which is an annoyance of pytest.  A perhaps cleaner way of doing this is through `request`:

```python
File Edit Options Buffers Tools Python Help                                                                                                 
import pytest                                                                                                                               
                                                                                                                                            
                                                                                                                                            
@pytest.fixture                                                                                                                             
def a():                                                                                                                                    
    return 'a'                                                                                                                              
                                                                                                                                            
@pytest.fixture                                                                                                                             
def b():                                                                                                                                    
    return 'b'                                                                                                                              
                                                                                                                                            
@pytest.fixture                                                                                                                             
def c():                                                                                                                                    
    return 'c'                                                                                                                              
                                                                                                                                            
@pytest.fixture(params=['a', 'b', 'c'])                                                                                                     
def myfixture(request, a, b, c):                                                                                                            
    if request.param == 'a':                                                                                                                
        yield a * 3                                                                                                                         
    elif request.param == 'b':                                                                                                              
        yield b * 3                                                                                                                         
    elif request.param == 'c':                                                                                                              
        yield c * 3                                                                                                                         
                                                                                                                                            
                                                                                                                                            
def test_this(myfixture):                                                                                                                   
    assert myfixture == 'aaa'                                                                                                                                                                                            
```

The `myfixture` fixture will make any test using it run 3 times, once for each param.  I think you can use this pattern to clean up the tests below and remove the `assert` inside the `assert` too